### PR TITLE
Allow to escape newlines and tabs in --set

### LINKED ISSUE #3251: Unable to --set with spaces or newlines
I'm trying to do:
```
helm upgrade kubernetes-dashboard stable/kubernetes-dashboard --reuse-values \
  --set 'ingress.annotations.nginx\.ingress\.kubernetes\.io/configuration-snippet=allow 10.20.30.40;\ndeny all;'
```

It doesn't work:
```
Error: This command needs 2 arguments: release name, chart path
```

NB I tried a lot of other things without success

### PR DESCRIPTION
**Pull request by sathieu**
Fixes #3251

### CODE REVIEW

**Review by bacongobbler — CHANGES_REQUESTED** [review]
I'm not 100% sure this is the right fix. Perhaps instead this should be fixed to parse `\\` as `\` and write tests around that test case. That would be able to handle any escape character moving forward, including Windows path-style arguments like `C:\\\\`. What do you think about that change?

I'm a little concerned about backwards compatibility with this change to the strvals parser too. Generally speaking, this would be the right fix for the parser, but I'm concerned that there may be a few charts out in the wild that are aware of this bug and have worked around it. By introducing this fix we could potentially break existing charts.

**Comment by sathieu** [comment]
@bacongobbler Current (before this PR) behavior is to replace every \\whatever by whatever. So \\\\ is replaced by \\. This behavior is kept. The only changes are for the documented escapes.

A better behavior IMO would be to reject any unknown escape sequence (i.e not in "stop" and not \\). But how to pass error down to the client?

Please accept this PR in the meantime.

File: pkg/strvals/parser_test.go [hunk 1]
```diff
@@ -255,6 +255,10 @@ func TestParseSet(t *testing.T) {
 			str:    "nested[1][1]=1",
 			expect: map[string]interface{}{"nested": []interface{}{nil, []interface{}{nil, 1}}},
 		},
+		{
+			str:    "escapes=hello\\n\\tworld",
+			expect: map[string]interface{}{"escapes": "hello\n\tworld"},
+		},
```
Comment (mattfarina) [inline pkg/strvals/parser_test.go]: Helm used a `\\` to escape things today. That means, `\\\n` would become `\n`. The test would pass without any other changes if the test were re-written as:

```
{
	str:    "escapes=hello\\n\\tworld",
	expect: map[string]interface{}{"escapes": "hello\n\tworld"},
},
```

Is there a reason for the other changes?

**Review by mattfarina — CHANGES_REQUESTED** [review]

File: pkg/strvals/parser_test.go (hunk 1 above)
Comment (mattfarina) [reply pkg/strvals/parser_test.go]: If we document the escape sequences I think we can close this.

Comment (sathieu) [reply pkg/strvals/parser_test.go]: It is unclear what you have changed in your test. Do you mean (three `\`):
```
{
	str:    "escapes=hello\\\n\\\tworld",
	expect: map[string]interface{}{"escapes": "hello\n\tworld"},
},
```

Comment (sathieu) [reply pkg/strvals/parser_test.go]: @mattfarina I don't know of any way to have a tab or newline currently (except passing the char directly).

**Pull request closed without merging**